### ISSUE TITLE
[MaterialDatePicker] Selection of Year, Month or Week only

### ISSUE THREAD

**Issue by Fishfield** [material-components/material-components-android#1961]
**Is your feature request related to a problem? Please describe.**
In our app, we have a statistics feature. The user can choose whether he wants to show statistics for either **a year, a month, or a week**. The user can then select which year, month, or week he wants to have displayed.

**Describe the solution you'd like**
The _MaterialDatePicker_ should have a feature, so the user can select either a year, a month, or a week and not just a custom time range that can't be restricted. 
Right now we have a custom solution in place which is tedious to maintain. It is error-prone and not handy to use

Thanks,

Best Thomas

**Comment by jagadishnallappa** [comment]
Aye. This feature needs to be implemented please! 

Thank you very much.

**Comment by rosghub** [comment]
Agreed this is desperately needed. The `CalendarView` used to be able to highlight the current week but the ability to do this has been deprecated since the age of material design. There's no built in way to do what OP wants in android and it's pretty odd given the use case is not uncommon.
